ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/metadata/MetadataPresenterImpl.java

```diff
@@ -1,29 +1,105 @@
 import org.iplantc.de.diskResource.client.presenters.callbacks.DiskResourceMetadataUpdateCallback;
+import org.iplantc.de.diskResource.client.views.metadata.dialogs.MetadataTemplateViewDialog;
+import org.iplantc.de.diskResource.client.views.metadata.dialogs.SelectMetadataTemplateDialog;
+import org.iplantc.de.resources.client.messages.I18N;
 
+import com.google.gwt.core.client.GWT;
 import com.google.gwt.user.client.rpc.AsyncCallback;
 import com.google.gwt.user.client.ui.HasOneWidget;
-
-import java.util.List;
+import com.sencha.gxt.widget.core.client.Dialog.PredefinedButton;
+import com.sencha.gxt.widget.core.client.box.ConfirmMessageBox;
+import com.sencha.gxt.widget.core.client.event.DialogHideEvent;
+import com.sencha.gxt.widget.core.client.event.DialogHideEvent.DialogHideHandler;
+import com.sencha.gxt.widget.core.client.event.HideEvent;
+import com.sencha.gxt.widget.core.client.event.SelectEvent;
 
 /**
  * @author jstroot
  */
 public class MetadataPresenterImpl implements MetadataView.Presenter {
+	
+	
+    private class CancelSelectHandler implements SelectEvent.SelectHandler {
+    	
+    	private MetadataTemplateViewDialog mdView;
+    	
+    	public CancelSelectHandler(MetadataTemplateViewDialog mdView) {
+			this.mdView = mdView;
+		}
+    	
+        @Override
+        public void onSelect(SelectEvent event) {
+        	mdView.hide();
+        }
+    }
+
+    private class OkSelectHandler implements SelectEvent.SelectHandler {
+    
+    	private MetadataTemplateViewDialog mdView;
+    	private boolean writable;
+    	private MetadataView.Presenter mdPresenter;
+    	
+    	public OkSelectHandler(boolean writable, MetadataView.Presenter mdPresenter, MetadataTemplateViewDialog mdView) {
+			this.writable = writable;
+			this.mdView = mdView;
+			this.mdPresenter = mdPresenter;
+		}
+    	
+        @Override
+        public void onSelect(SelectEvent event) {
+            if(!writable){
+                return;
+            }
+
+            if (!mdView.isValid()) {
+                ConfirmMessageBox cmb = new ConfirmMessageBox("Error", "Metadata incomplete");
+                cmb.addDialogHideHandler(new DialogHideHandler() {
+                    
+                    @Override
+                    public void onDialogHide(DialogHideEvent event) {
+                        if (event.getHideButton().equals(PredefinedButton.YES)) {
+                            mdView.mask(I18N.DISPLAY.loadingMask());
+                            ArrayList<DiskResourceMetadata> mdList = mdView.getMetadataFromTemplate();
+                            view.updateMetadataFromTemplateView(mdList);
+                        }
+                        
+                    }
+                });
+                cmb.show();
+            } else {
+                mdView.mask(I18N.DISPLAY.loadingMask());
+                ArrayList<DiskResourceMetadata> mdList = mdView.getMetadataFromTemplate();
+                view.updateMetadataFromTemplateView(mdList);
```

Comment:
This is just a minor suggestion, but since three lines of code are repeated on lines 78 through 80 and 87 through 89, it might be helpful to extract them into a private method inside OKSelectHandler.